### ISSUE TITLE
More of a Question than an issue

### ISSUE THREAD

**Issue by mreverettb1** [bullhorn/dataloader#420]
Can Bullhorn Dataloader create new company records if no match is made?

I am trying to import (or load) new company contacts to companies. Some of them are new and the companies are not set up in Bullhorn.  I am getting non-match error on the contacts that their companies are not in Bullhorn, if the company is there, I do not get any errors, and the contact imports successfully.  

### Steps to Reproduce

1. Select CSV file, Choose Entity, Map Columns, Dup check. 
2. Let run, and get return results. 
3. return results:  Error:  Record Not Found - Message:  Cannot find clientCorporation with Name "companyname" 

##### Expected behavior:
1. Add companies that do not match 
2. Companies that were in,  the dataloader added the records

[What you expect to happen]

##### Actual behavior:

[What actually happens]

##### Reproduces how often:
Each time I run it

[What percentage of the time does it reproduce?]
when no match is found.

### Additional Information

1. The `dataloader.properties` file (minus login info)
3. The CSV input file (smallest possible file that reproduces the issue)
4. The results file(s)
5. The log file

**Comment by ndickerson** [comment]
In Data Loader, only the entity being loaded will perform an Upsert (Update if present, Insert if not present). In order to ensure that companies that do not exist in Bullhorn do exist, you'll need to first load ClientCorporation records, and provide the information that you want to be in the ClientCorporation if it does not exist. The only required field is `name`, so that the company has a name to display for it. Then when you load your Contact records, it will Upsert the contacts, and all companies should already be present. Does that answer your question?

**Comment by mreverettb1** [comment]
Hi Nathan,

 

Yes, that explains it perfect.  So I should do a first run to get the companies in,  then do a second run to get the contacts in?  I wrote a nice little extracting script that pulls data and formats it into a very easy read for the recruiters, and Bullhorn.   The problem with Bullhorn is that I can import the contacts, however it will not update the existing data. That is why I am trying to get the Bullhorn Dataloader going, as it will update.

 

Do you know if there is away to upload in a single pass?

 

Thank you  very much for the reply and insight on this new project I Am working on!!  😊

 

Best

Everett

**Comment by mreverettb1** [comment]
Thank you very much!!  one other question if you don't mind,  when importing new contacts,  can you attach them to a company that is in BH?  (I am trying to do that by adding the clientcorporation) however it's failing to add......but will update?
Thanks again
Best
Everett

**Comment by ndickerson** [comment]
Yes, you can add contacts to a clientCorporation by filling out the clientCorporation column on a new/updated clientContact csv file. You can match on any field in the clientCorporation you wish, such as id or name.

**Comment by mreverettb1** [comment]
Hey Nathan,  So I have not been successful getting the contacts to add to the companies.  I run the CSV file through to get the companies into and or updated in BH.  I then take the same CSV file and import the contacts, Mapping the company field and the results I get are:  on the contacts record the company shows as "Imported Contact".  Would you know why this is happening???  
Best
Everett
